src/components/inputModal/index.tsx

```diff
@@ -1,62 +1,93 @@
 import { MODAL_TYPE } from "utils/enums";
-import InputTextField from "components/inputModal/InputTextField";
-import InputSelectField from "components/inputModal/InputSelect";
-import {grades, subjects} from "utils/index";
+import InputTextField from "components/inputModal/inputTextField";
+import InputSelectField from "components/inputModal/inputSelect";
+import { grades, subjects } from "utils/index";
+import { IStudentRaw } from "state/ducks/students/types";
+import { ActionType } from "typesafe-actions";
+import { addStudent, updateStudent } from "state/ducks/students/actions";
 interface IModalProps {
   visible: boolean;
-  setVisible: Function;
-  // isEdit: boolean;
+  setVisible: (x: boolean) => void;
+  addStudent: (payload: IStudentRaw) => ActionType<typeof addStudent>;
+  updateStudent: (payload: IStudentRaw) => ActionType<typeof updateStudent>;
+  studentData: IStudentRaw;
+  setStudentData: (std: IStudentRaw) => void;
 }
 
 const StudentInputModal: React.FC<IModalProps> = (props) => {
-  const initState = {
+  const blankForm = {
     name: "",
+    marks: 0,
     subject: "",
-    marks: "",
-    grades: "",
+    grade: "",
   };
 
-  // const modalType = props?.isEdit ? MODAL_TYPE.EDIT : MODAL_TYPE.ADD;
-  const modalType = MODAL_TYPE.ADD;
+  const modalType = props?.studentData?._id ? MODAL_TYPE.EDIT : MODAL_TYPE.ADD;
 
   const {
-    register,
     handleSubmit,
     reset,
     control,
-    formState: { errors, isSubmitSuccessful },
-  } = useForm({
+    formState: { errors },
+  } = useForm<IStudentRaw>({
     mode: "onTouched",
     reValidateMode: "onChange",
     resolver: yupResolver(schema),
-    defaultValues: initState,
+    defaultValues: useMemo(() => {
+      return props.studentData || blankForm;
+    }, [props.studentData]),
   });
 
   useEffect(() => {
-    if (isSubmitSuccessful) {
-      reset();
-      console.log("Reset form!");
+    reset(props.studentData);
+  }, [props.studentData]);
+
+  const onSubmit = (values: IStudentRaw) => {
+    reset(blankForm);
+    const date = new Date();
+    const formattedDate = date.toLocaleDateString("en-GB", {
+      day: "2-digit",
+      month: "2-digit",
+      year: "numeric",
+    });
+    const formattedTime = date.toLocaleTimeString("en-US", {
+      hour: "2-digit",
+      minute: "2-digit",
+      hour12: true,
+    });
+    if (modalType === MODAL_TYPE.EDIT) {
+      props.updateStudent({
+        ...values,
+        _id: props?.studentData._id,
+        date: formattedDate,
+        time: formattedTime,
+      });
+    } else {
+      props.addStudent({ ...values, date: formattedDate, time: formattedTime });
     }
-  }, [isSubmitSuccessful, reset, props.visible]);
+    props.setStudentData(blankForm);
+    props.setVisible(false);
+  };
 
-  const onSubmit = (values: object) => {
-    console.log("Values:::", values);
+  const handleClose = () => {
     props.setVisible(false);
+    props.setStudentData(blankForm);
```

Comment:
I think studentData should have `null` instead of `blankForm` as states should not have misleading or fake data.